File: app/src/main/java/com/example/rickmortymvvm/list/viewmodel/PresentationCharacterListAction.java
package com.example.rickmortymvvm.list.viewmodel;

import com.example.rickmortymvvm.models.Character;

public abstract class PresentationCharacterListAction {
    public static class GoToInfo extends PresentationCharacterListAction {
        private final Character character;

        public GoToInfo(Character character) {
            this.character = character;
        }

        public Character getCharacter() {
            return character;
        }
    }

    public static class Finish extends PresentationCharacterListAction {

    }
}

File: app/src/main/java/com/example/rickmortymvvm/list/viewmodel/PresentationCharacterListAction.kt
package com.example.rickmortymvvm.list.viewmodel

import com.example.rickmortymvvm.models.Character

sealed class PresentationCharacterListAction {
    data class GoToInfo(val character: Character) : PresentationCharacterListAction()
    object Finish : PresentationCharacterListAction()
}
